mlrun/api/main.py

```diff
@@ -155,23 +161,29 @@ async def startup_event():
     await _initialize_singletons()
 
     # periodic cleanup is not needed if we're not inside kubernetes cluster
-    if get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster():
+    if (
+        get_k8s_helper(silent=True).is_running_inside_kubernetes_cluster()
+        and config.httpdb.state == mlrun.api.schemas.APIStates.online
+    ):
         _start_periodic_cleanup()
         _start_periodic_runs_monitoring()
 
 
 @app.on_event("shutdown")
 async def shutdown_event():
-    get_project_member().shutdown()
+    if get_project_member():
+        get_project_member().shutdown()
     cancel_all_periodic_functions()
-    await get_scheduler().stop()
+    if get_scheduler():
+        await get_scheduler().stop()
 
 
 async def _initialize_singletons():
```

Comment:
This too is called in the `startup_event` flow, which means that if we're starting up in a state that's not `online`, it should be called again to start the scheduler and everything. How is this handled?

Reply:
was indeed missing - added